sphinxcontrib/needs/diagrams_common.py

```diff
@@ -94,16 +94,16 @@ def collect_diagram_attributes(self):
             caption = self.arguments[0]
 
         collected_diagram_options = {
-            'show_legend': True if self.options.get("show_legend", False) is None else False,
```

Comment:
@danwos  what is this meant to be checking?

`self.options.get("show_legend", False) is None` will *always* evaluate to `False` since you're comparing a bool to `None`, right?

also i've simplified
```python
True if self.options.get("show_legend", False) is None else False
```
to
```python
self.options.get("show_legend", False) is None
```
which can be further simplified to
```python
not self.options.get("show_legend")
```
on the assumption that `self.options.get("show_legend", False) is None` is supposed to be `self.options.get("show_legend", None) is None`

but that seems an inversion of the intention? ie. this will be true if "show_legend" is not set, or is set to None, and false if "show_legend" is set to anything except `None`.

is this meant to just be


```python
self.options.get("show_legend", False)
```

???

Reply:
i've updated this to what i *think* this was meant to be checking all along...

Reply:
i guess i did break the logic here-

```
sphinxcontrib.needs.layout.SphinxNeedLayoutException: Given layout "" is unknown for need FEATURE_3. Registered layouts are: test ,clean ,clean_l ,clean_lp ,clean_r ,clean_rp ,complete ,focus ,focus_f ,focus_l ,focus_r ,debug ,example ,github
```

should these be


```python
bool(self.options.get("show_legend", False))
```

??

i still think the problem lies with `self.options` being 'stringly'-typed

Reply:
Yeah there is a lot of voodoo.
Is far as I remember the flag handling in sphinx is a little bit strange.

Example:
```
.. need::
   :show_options:   #Simple flag, no Value!
```
To figure out, if ``show_options`` is set, it must be part of ``self.options``, but it's value will be ``None``.
If it is not set, it's not part of ``self.options``.
So we can't just copy its value, as it is not set.
So the original ``show_legend`` from sphinx is not a bolean, my code tries to make it to one (in a really too complex way.)

I think ``True if self.options.get("show_legend", False) is None else False`` can be transformed to ``"show_legend in self.options``.

But I have already recognized that your logic is "much stronger" as mine ;)

Reply:
ah I see, thankyou. I was thinking the check was trying to see if the value was set *or* was `False`, but if it's sufficient to check whether it exists at all then `"show_legend" in self.options` is probably the way to go!

> But I have already recognized that your logic is "much stronger" as mine ;)

Dunno about that. sometimes you just need fresh eyes on something